internal/querynode/metrics_info.go

```diff
@@ -20,11 +20,34 @@ import (
 	"context"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
 	"github.com/milvus-io/milvus/internal/proto/milvuspb"
 	"github.com/milvus-io/milvus/internal/util/metricsinfo"
 	"github.com/milvus-io/milvus/internal/util/typeutil"
 )
 
+//getComponentConfigurations returns the configurations of queryNode matching req.Pattern
+func getComponentConfigurations(ctx context.Context, req *internalpb.ShowConfigurationsRequest) *internalpb.ShowConfigurationsResponse {
+	prefix := "querynode."
+	matchedConfig := Params.QueryNodeCfg.Base.GetByPattern(prefix + req.Pattern)
+	configList := make([]*commonpb.KeyValuePair, len(matchedConfig))
```

Comment:
shall be `make([]*commonpb.KeyValuePair, 0, len(matchedConfig)`, otherwise result will be:
```{nil, nil, nil, nil..., result1, result2, result3, ...}```